src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -2172,6 +2172,22 @@ impl<'a, 'b> Mul<&'a Decimal> for &'b Decimal {
     }
 }
 
+impl<'a, 'b> Div<&'a Decimal> for &'b Decimal {
+    type Output = Option<Res<Decimal>>;
+
+    fn div(self, rhs: &'a Decimal) -> Self::Output {
```

Comment:
I believe you can change the implementation of `Decimal::div` to accept references, instead of duplicating a new one while preserving the old one.